staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -693,17 +693,35 @@ func nodeAddressesChangeDetected(addressSet1, addressSet2 []v1.NodeAddress) bool
 	return false
 }
 
+// ensureNodeProvidedIPExists will check if the nodeIP suggested by the user is
+// present and move it to the front of the list of node IP addresses if so
 func ensureNodeProvidedIPExists(node *v1.Node, nodeAddresses []v1.NodeAddress) (*v1.NodeAddress, bool) {
```

Comment:
This function probably needs to be renamed now too, because we're no longer just ensuring the node IP is there, but we're also shuffling the order.